tests/device_plugin/mod.rs

```diff
@@ -0,0 +1,195 @@
+pub(crate) mod v1beta1 {
+    pub const API_VERSION: &str = "v1beta1";
+    tonic::include_proto!("v1beta1");
+}
+#[path = "../grpc_sock/mod.rs"]
+pub mod grpc_sock;
+use futures::Stream;
+use std::path::Path;
+use std::pin::Pin;
+use tokio::sync::mpsc;
+use tonic::{Request, Response, Status};
+use v1beta1::{
+    device_plugin_server::{DevicePlugin, DevicePluginServer},
+    registration_client, AllocateRequest, AllocateResponse, ContainerAllocateResponse, Device,
+    DevicePluginOptions, Empty, ListAndWatchResponse, Mount, PreStartContainerRequest,
+    PreStartContainerResponse, PreferredAllocationRequest, PreferredAllocationResponse,
+    RegisterRequest, API_VERSION,
+};
+
+/// Mock Device Plugin for testing the DeviceManager. Reports its list of devices.
+struct MockDevicePlugin {
+    // Devices that are advertised
+    devices: Vec<Device>,
+}
+
+#[async_trait::async_trait]
+impl DevicePlugin for MockDevicePlugin {
+    async fn get_device_plugin_options(
+        &self,
+        _request: Request<Empty>,
+    ) -> Result<Response<DevicePluginOptions>, Status> {
+        unimplemented!();
+    }
+
+    type ListAndWatchStream =
+        Pin<Box<dyn Stream<Item = Result<ListAndWatchResponse, Status>> + Send + Sync + 'static>>;
+    async fn list_and_watch(
+        &self,
+        _request: Request<Empty>,
+    ) -> Result<Response<Self::ListAndWatchStream>, Status> {
+        println!("list_and_watch entered");
+        let (kubelet_update_sender, kubelet_update_receiver) = mpsc::channel(3);
+        let devices = self.devices.clone();
+        tokio::spawn(async move {
+            kubelet_update_sender
+                .send(Ok(ListAndWatchResponse { devices }))
+                .await
+                .unwrap();
+            loop {
+                // List and watch should not end
+                tokio::time::sleep(std::time::Duration::from_secs(5)).await;
```

Comment:
Super nit: I think we use the uint max value to basically never wake up, but it really isn't too big of a deal in a test

Reply:
fixed! ec47ee860d808c84f10a4a64381a399688f356b4